### PR TITLE
Allow upgrades and downgrades.

### PR DESCRIPTION
**Pull request by PEConn**
Hey Andre and Michael, PTAL.

This work corresponds to [this bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1169203)

### CODE REVIEW

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/MethodData.java [hunk 1]
```diff
@@ -103,4 +119,37 @@ public static MethodData fromIntent(Intent intent) {
...
+            case "immediateWithoutProration":
+                return BillingFlowParams.ProrationMode.IMMEDIATE_WITHOUT_PRORATION;
+            case "immediateWithTimeProration":
+                return BillingFlowParams.ProrationMode.IMMEDIATE_WITH_TIME_PRORATION;
+            case "unknownSubscriptionUpgradeDowngradePolicy":
+                return BillingFlowParams.ProrationMode.UNKNOWN_SUBSCRIPTION_UPGRADE_DOWNGRADE_POLICY;
+            default:
+                return null;
+        }
+//        try {
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/MethodData.java:148]: Are those are left-over comments? Do we want to remove them?

**Review by andreban — APPROVED** [review]
LGTM, one nit around some commented-out code.

File: playbilling/src/test/java/com/google/androidbrowserhelper/playbilling/provider/MethodDataTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    public void fromJson_purchaseToken() {
+        String json = "{ 'sku' = 'mySku', 'purchaseToken' = 'abc123' }"
+                .replace('\'', '\"');
+        MethodData data = MethodData.fromJson(json);
+
+        assertMethodData(data, "mySku", null, "abc123", null);
+    }
+
+    @Test
+    public void fromJson_prorationMode() {
```
Comment (mvano) [inline playbilling/src/test/java/com/google/androidbrowserhelper/playbilling/provider/MethodDataTest.java:57]: You might test that no match returns null.

**Review by mvano — APPROVED** [review]
Ditto Andre's comments.

**Pull request merged**